Issue 3188. Discharge Papers

RomanGolchuk opened the pull request:
#3188

abraun-ml commented on packages/utils/lib/fhir/constants.ts:
```diff
@@ -340,47 +357,47 @@ export interface ListConfig {
-    title: 'labs',
+    title: BUCKET_NAMES.LABS,
     display: 'Labs',
     documentTypeCode: [
```
Not something to change here, but a reminder to us @szaccagni we will need to update this with our new abn and ereq documents

abraun-ml commented on packages/utils/lib/utils/pdf.ts:
```diff
@@ -290,3 +290,83 @@ export async function uploadPDF(
+  }
+}
+
+async function getPresignedFileUrl(url: string, token: string): Promise<string | undefined> {
```
Any reason not to use the getPresignedFileUrl helper that exists in `[apps/ehr/src/helpers/files.helper](https://github.com/masslight/ottehr/blob/615a2f8cab5d86c448965e0f8fc1bec022d4e7e4/apps/ehr/src/helpers/files.helper.ts#L1)`?

abraun-ml commented on packages/zambdas/src/ehr/create-discharge-summary/index.ts:
```diff
@@ -0,0 +1,131 @@
+
+  console.log('Chart data received');
+  const pdfInfo = await composeAndCreateDischargeSummaryPdf(
+    { chartData, additionalChartData, radiologyData },
```
I think you will want to include in house lab results here (because you should have results on most if not all), and probably external lab orders (you will not yet have results)

abraun-ml commented on packages/zambdas/src/shared/pdf/discharge-summary-pdf.ts:
```diff
@@ -0,0 +1,573 @@
+  if (data.currentMedications?.length || data.currentMedicationsNotes?.length) drawCurrentMedications();
+  if (data.allergies?.length || data.allergiesNotes?.length) drawAllergies();
+  if (Object.values(data.vitals || {}).some((val) => !!val)) drawVitalsSection();
+  if (data.inHouseLabs?.length) drawInHouseLabs();
```
I don't know if it is a requirement, but I think you may also want to include the external lab orders (you won't have results). I could imagine a patient might want to know what tests were ordered. @kjames14 do you know what the requirement is?

kjames14 replied:
I would think yes, but I’ll defer to @GambleHeffernan since she worked on this feature

kjames14 replied:
Spoke with Tor, we should include external labs

abraun-ml commented on packages/zambdas/src/shared/pdf/external-labs-label-pdf.ts:
```diff
@@ -187,11 +186,11 @@ async function createExternalLabsLabelPDFHelper(
     secrets,
     fileName,
-    bucketName: UPLOAD_BUCKET_NAME,
+    bucketName: BUCKET_NAMES.VISIT_NOTES,
```
I actually think this could be the labs bucket

RomanGolchuk replied:
Great catch!

abraun-ml commented on packages/zambdas/src/shared/pdf/external-labs-order-form-pdf.ts:
```diff
@@ -31,7 +31,7 @@ export async function createExternalLabsOrderFormPDF(
 
   console.debug(`Created external labs order form pdf bytes`);
-  const bucketName = 'visit-notes';
+  const bucketName = BUCKET_NAMES.VISIT_NOTES;
```
same here, labs bucket

abraun-ml commented on packages/zambdas/src/shared/pdf/labs-results-form-pdf.ts:
```diff
@@ -1015,7 +1016,7 @@ async function createLabsResultsFormPDF(
 
   console.debug(`Created external labs order form pdf bytes`);
-  const bucketName = 'visit-notes';
+  const bucketName = BUCKET_NAMES.VISIT_NOTES;
```
labs bucket

abraun-ml commented on packages/zambdas/src/shared/pdf/visit-label-pdf.ts:
```diff
@@ -164,11 +162,11 @@ async function createVisitLabelPDFHelper(
     secrets,
     fileName,
-    bucketName: UPLOAD_BUCKET_NAME,
+    bucketName: BUCKET_NAMES.VISIT_NOTES,
```
Just to be clear, this one can stay visit notes :)

szaccagni commented on packages/zambdas/src/shared/pdf/discharge-summary-pdf.ts:
```diff
@@ -0,0 +1,573 @@
+  const { date: dischargedDate, time: dischargeTime } = formatDateToMDYWithTime(admitterParticipant?.period?.end) ?? {};
+  const dischargeDateTime = dischargedDate && dischargeTime ? `${dischargedDate} at ${dischargeTime}` : undefined;
+
+  return {
```
i think external labs is missing here

kjames14 replied:
Spoke with Tor, we should include external labs

alexwillingham commented:
Hi Roman. I was late to the party reviewing this. It looks like the PR has got a lot more in it than just a discharge PDF feature. I think you have got a feature branch from Aykhan accidentally included into your PR?

RomanGolchuk commented:
@alexwillingham, thanks for pointing that out!

I'm not sure how Aykhan's commits ended up in my PR — they weren't present in my local branch initially. I’ve just rebased the branch onto release/1.14 and removed the unrelated commits from the PR.

Now it should only contain the relevant changes for the discharge summary feature.

VladMstv commented:
@abraun-ml @szaccagni @alexwillingham can you please approve if everything was resolved?

abraun-ml commented on packages/zambdas/src/shared/pdf/discharge-summary-pdf.ts:
```diff
@@ -0,0 +1,715 @@
+    pdfClient.drawText('In-House Labs', textStyles.subHeader);
+    if (data.inHouseLabs?.orders.length) {
+      pdfClient.drawText('Orders:', textStyles.subHeader);
+      data.inHouseLabs?.orders.forEach((lab) => {
```
This is a readability nit. A lot of the labs terminology is overloaded and so it becomes hard to follow pretty fast. Your other forEach loops have this issue as well.

lab -> Laboratory
test or testItem -> actual test performed
order -> set of tests (can be singleton)

You probably want order or testItem here
```suggestion
      data.inHouseLabs?.orders.forEach((order) => {
```
